packages/design-system/src/components/Dropdown/Dropdown.tsx

```diff
@@ -89,6 +93,9 @@ const Dropdown = forwardRef(
 								{...menu}
 								key={`${label}-${index}`}
 								id={`${label}-${index}`}
+								onClick={() => {
+									menu.hide();
```

Comment:
We could still have `onClick` for this link though, if I'm not mistaken.

Reply:
That's not what indicate the type we want for the link
https://github.com/Talend/ui/blob/2119fdf1432dfaeb521dccd8a610a2e5ab3ac72a/packages/design-system/src/components/Dropdown/Dropdown.tsx#L19

But i can alter it in this PR if you want so :)

Reply:
Yup `LinkableType = Omit<AnchorHTMLAttributes<HTMLAnchorElement>, 'style'>...`

Reply:
And `AnchorHTMLAttributes<HTMLAnchorElement>` can take `onClick`

Reply:
I explored the typing too and it doesn't seem possible

Reply:
<img width="463" alt="image" src="https://user-images.githubusercontent.com/10761073/167089016-b54f77d0-37fe-42b9-bb7f-6f8b66a509a9.png">

Reply:
`onClick` is valid on all HTMLAttributes. The screenshot only shows the subset that's valid only for Anchor.
Links can have `onClick` handlers (for tracking, for this instance etc...), but buttons can't have `href` attributes.

Reply:
just tried out and it works, so you must add the entry?.onClick here

Reply:
I can add the on click if wanted so
I don't have really strong strong opinion about this.
I would tend to think like Jimmy and if you want to have a callback, it's more a button to place but i can add it in order to be less restrictive

Reply:
(added 🥳 )

Reply:
Let's not get confused here :D 

- If you want a callback, use a button 
- If you want navigation, use a link. If you need a callback upon navigation, HTML says you can.